src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -1253,6 +1272,8 @@ export class TabsTitleControl extends TitleControl {
 				this.layoutScheduled.clear();
 			});
 		}
+
+		return this.getDimensions().height;
```

Comment:
As you can see from the code above, the layout is scheduled `async` to prevent perf issues. As such, you return a dimension here from a time before the layout is actually applied. This is one of the issues I talked about earlier, we cannot return the height until we actually applied the CSS classes. I fear we probably have to switch to a sync layout when multi-line tabs are enabled.

Reply:
Ah true, I overlooked that. I'm not very familiar with how the delegation of the layout to a new animation frame works, so I may need some assistance on the implementation here.